tests/gopherjsvendored_test.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Don't run this file directly. It's executed as part of TestGopherJSCanBeVendored.
+
+set -e
+
+tmp=$(mktemp -d "${TMPDIR:-/tmp}/gopherjsvendored_test.XXXXXXXXXX")
+
+cleanup() {
+    rm -rf "$tmp"
+    exit
+}
+
+trap cleanup EXIT SIGHUP SIGINT SIGTERM
+
+# Make a hello project that will vendor GopherJS.
+mkdir -p "$tmp/src/example.org/hello"
+echo 'package main
+
+import "github.com/gopherjs/gopherjs/js"
+
+func main() {
+    js.Global.Get("console").Call("log", "hello using js pkg")
+}' > "$tmp/src/example.org/hello/main.go"
+
+# Vendor GopherJS and its dependencies into hello project.
+mkdir -p "$tmp/src/example.org/hello/vendor/github.com/gopherjs" \
+         "$tmp/src/example.org/hello/vendor/github.com/fsnotify" \
+         "$tmp/src/example.org/hello/vendor/github.com/kisielk" \
+         "$tmp/src/example.org/hello/vendor/github.com/neelance" \
+         "$tmp/src/example.org/hello/vendor/github.com/shurcooL" \
+         "$tmp/src/example.org/hello/vendor/github.com/spf13" \
+         "$tmp/src/example.org/hello/vendor/golang.org/x"
+cp -r $(go list -e -f '{{.Dir}}' github.com/gopherjs/gopherjs)  "$tmp/src/example.org/hello/vendor/github.com/gopherjs/gopherjs"
+cp -r $(go list -e -f '{{.Dir}}' github.com/fsnotify/fsnotify)  "$tmp/src/example.org/hello/vendor/github.com/fsnotify/fsnotify"
+cp -r $(go list -e -f '{{.Dir}}' github.com/kisielk/gotool)     "$tmp/src/example.org/hello/vendor/github.com/kisielk/gotool"
+cp -r $(go list -e -f '{{.Dir}}' github.com/neelance/sourcemap) "$tmp/src/example.org/hello/vendor/github.com/neelance/sourcemap"
+cp -r $(go list -e -f '{{.Dir}}' github.com/shurcooL/httpfs)    "$tmp/src/example.org/hello/vendor/github.com/shurcooL/httpfs"
+cp -r $(go list -e -f '{{.Dir}}' github.com/spf13/cobra)        "$tmp/src/example.org/hello/vendor/github.com/spf13/cobra"
+cp -r $(go list -e -f '{{.Dir}}' github.com/spf13/pflag)        "$tmp/src/example.org/hello/vendor/github.com/spf13/pflag"
+cp -r $(go list -e -f '{{.Dir}}' golang.org/x/crypto)           "$tmp/src/example.org/hello/vendor/golang.org/x/crypto"
+cp -r $(go list -e -f '{{.Dir}}' golang.org/x/sys)              "$tmp/src/example.org/hello/vendor/golang.org/x/sys"
+cp -r $(go list -e -f '{{.Dir}}' golang.org/x/tools)            "$tmp/src/example.org/hello/vendor/golang.org/x/tools"
```

Comment:
If GopherJS's dependencies are changed, we should update these items, right? At least, should we leave a comment?

Reply:
Yeah. Perhaps this script can be improved in the future to infer dependencies more automatically (e.g., with `go list -f '{{join .Deps "\n"}}' github.com/gopherjs/gopherjs/...` or so).

There's a comment on top saying `# Vendor GopherJS and its dependencies into hello project.`, is it not clear?

Reply:
Feels like we should make this change now; it's brittle having these hard-coded, particularly when using `go list` is something that we're already doing elsewhere and is easy to make work in this situation.

Reply:
Agreed with @myitcv . This doesn't look so hard.

Reply:
Done.